src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -36,82 +29,41 @@ public class LuceneIndexWriter : IDisposable
 
         private readonly IndexWriter.MaxFieldLength maxFieldLength;
 
-        private int currentNumberOfWrites;
-
         private readonly IndexWriter.IndexReaderWarmer _indexReaderWarmer;
 
-        private int changesSinceCommit;
-
-        private readonly Field forceCommitField = new Field("__dummy_force_index_write", "forced", Field.Store.NO, Field.Index.NOT_ANALYZED_NO_NORMS);
-        private readonly Term forceCommitTerm = new Term("__dummy_force_index_write", "forced");
-        private readonly global::Lucene.Net.Documents.Document forceCommitDoc;
-        private DateTime lastCommitDataStoreTime;
-
-        public Directory Directory
-        {
-            get
-            {
-                if (indexWriter != null)
-                    return indexWriter.Directory;
-
-                return null;
-            }
-        }
-
-        public Analyzer Analyzer
-        {
-            get
-            {
-                if (indexWriter != null)
-                    return indexWriter.Analyzer;
+        public Directory Directory => indexWriter?.Directory;
 
-                return null;
-            }
-        }
+        public Analyzer Analyzer => indexWriter?.Analyzer;
 
-        public LuceneIndexWriter(Directory d, Analyzer a, IndexDeletionPolicy deletionPolicy, IndexWriter.MaxFieldLength mfl, int maximumNumberOfWritesBeforeRecreate, IndexWriter.IndexReaderWarmer indexReaderWarmer)
+        public LuceneIndexWriter(Directory d, Analyzer a, IndexDeletionPolicy deletionPolicy, IndexWriter.MaxFieldLength mfl, IndexWriter.IndexReaderWarmer indexReaderWarmer)
         {
             directory = d;
             analyzer = a;
             indexDeletionPolicy = deletionPolicy;
             maxFieldLength = mfl;
             _indexReaderWarmer = indexReaderWarmer;
-            this.maximumNumberOfWritesBeforeRecreate = maximumNumberOfWritesBeforeRecreate;
-
-            forceCommitDoc = new global::Lucene.Net.Documents.Document();
-            forceCommitDoc.Add(forceCommitField);
 
-            RecreateIfNecessary(force: true);
+            RecreateIndexWriter();
```

Comment:
Are you sure that this re-create is cheap? I think that Lucene needs to read a bunch of stuff that it holds in memory whenever it open the writer.